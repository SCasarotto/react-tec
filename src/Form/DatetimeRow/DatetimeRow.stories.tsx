import React from 'react';

import { Story, Meta } from '@storybook/react';

import { DatetimeRow, DatetimeRowProps } from './DatetimeRow';

export default {
  title: 'Form/DatetimeRow',
  component: DatetimeRow,
  argTypes: {},
} as Meta;

const Template: Story<DatetimeRowProps> = (args) => <DatetimeRow {...args} />;

export const Default = Template.bind({});
Default.args = {
  labelForKey: 'SomeUniqueKey',
  title: 'Datetime Row',
};
