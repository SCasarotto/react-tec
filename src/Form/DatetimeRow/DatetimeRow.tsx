import React from 'react';

import ReactDatePicker from 'react-datepicker';

import { DatetimeInput, DatetimeInputProps } from '../DatetimeInput';
import { FormRow, FormRowCustomProps } from '../FormRow';
import { Label, LabelCustomProps } from '../Label';

export interface DatetimeRowProps
  extends FormRowCustomProps,
    LabelCustomProps,
    DatetimeInputProps {
  labelForKey: string;
}
/**
 *
 * A `Row`, `Label` and `DatetimeInput`. In addition to the props below, reference all props for `DatetimeInput`.
 *
 */
export const DatetimeRow = React.forwardRef<
  // TODO: Improve generic type and entend with generic like Select
  ReactDatePicker<any>,
  DatetimeRowProps
>((props, ref) => {
  const { rowSize, last, className = '', title, labelForKey, ...rest } = props;

  return (
    <FormRow
      rowSize={rowSize}
      last={last}
      className={`tec-datetimeRow ${className}`}
    >
      {title && (
        <Label
          htmlFor={labelForKey}
          required={rest.required}
          disabled={rest.disabled}
          className='tec-datetimeRowTitle'
        >
          {title}
        </Label>
      )}
      <DatetimeInput
        id={labelForKey}
        {...rest}
        className='tec-datetimeRowDateInput'
        ref={ref}
      />
    </FormRow>
  );
});
