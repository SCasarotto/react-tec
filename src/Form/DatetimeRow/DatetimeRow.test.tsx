import React from 'react';

import { render, waitFor } from '@testing-library/react';
import DatePicker from 'react-datepicker';

import { DatetimeRow } from './DatetimeRow';

describe('DatetimeRow', () => {
  // Renders Without Error
  it('matches snapshot', () => {
    const { asFragment } = render(
      <DatetimeRow labelForKey='myDatetimeRow' onChange={jest.fn()} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  // Class Name
  it('accepts custom className', () => {
    const { container } = render(
      <DatetimeRow
        className='myClassName'
        labelForKey='myDatetimeRow'
        onChange={jest.fn()}
      />,
    );
    expect(container.firstChild).toHaveClass('myClassName');
  });

  // Ref
  it('renders with ref', async () => {
    const ref: { current: DatePicker<any> | null } = { current: null };
    render(
      <DatetimeRow
        labelForKey='myDatetimeRow'
        onChange={jest.fn()}
        ref={ref}
      />,
    );
    waitFor(() => expect(ref.current).toBeTruthy());
  });
});
