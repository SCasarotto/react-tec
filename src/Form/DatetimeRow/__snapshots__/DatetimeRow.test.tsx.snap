// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatetimeRow matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  display: inline-block;
  vertical-align: top;
  margin-bottom: 10px;
  text-align: left;
  box-sizing: border-box;
  padding-right: 0px;
  width: 100%;
}

<div
    class="c0 tec-formRow tec-datetimeRow "
  >
    <div
      class="react-datepicker-wrapper"
    >
      <div
        class="react-datepicker__input-container"
      >
        <input
          autocomplete="off"
          class="tec-datetimeInput tec-datetimeRowDateInput"
          id="myDatetimeRow"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
